src/Data/Aeson/TH.hs

```diff
@@ -328,7 +332,9 @@ consToValue :: ToJSONFun
             -> Q Exp
 
 consToValue _ _ _ _ [] =
-    [| \x -> x `seq` error "case: V1" |]
+    -- If we put the expression into the TH-generated code directly,
+    -- that makes it require EmptyCase at use-sites too
+    [| emptyCase |]
```

Comment:
why not `[| \x -> case x of {} |]`?

It's fine to require users to enable `EmptyCase`.

Reply:
Is there any benefit in doing that?

Reply:
Yes, `-ddump-splices` doesn't look suspicious.

Reply:
Done